animation_nodes/nodes/mesh/generation/line.py

```diff
@@ -1,18 +1,46 @@
 import bpy
+from bpy.props import *
+from .... data_structures import Mesh
 from .... base_types import AnimationNode
-from .... algorithms.mesh_generation.line import getLineMesh
+from .... algorithms.mesh_generation.line import getLineMesh, getLinesMesh
+
+lineModeItems = [
+    ("STARTEND", "Start-End", "Line from start-end", "NONE", 0),    
+    ("POINTS", "Points", "Line from vector list", "NONE", 1)
+]
 
 class LineMeshNode(bpy.types.Node, AnimationNode):
     bl_idname = "an_LineMeshNode"
     bl_label = "Line Mesh"
 
+    lineMode: EnumProperty(name = "Line Mode", default = "STARTEND",
+        items = lineModeItems, update = AnimationNode.refresh)
+
     def create(self):
-        self.newInput("Vector", "Start", "start")
-        self.newInput("Vector", "End", "end", value = [5, 0, 0])
-        self.newInput("Integer", "Steps", "steps", value = 2, minValue = 2)
+        if self.lineMode == "STARTEND":
+            self.newInput("Vector", "Start", "start")
+            self.newInput("Vector", "End", "end", value = [5, 0, 0])
+            self.newInput("Integer", "Steps", "steps", value = 2, minValue = 2)
+        elif self.lineMode == "POINTS":
+            self.newInput("Vector List", "Points", "points")
+            self.newInput("Boolean", "Cyclic", "cyclic", value = False)
 
         self.newOutput("Mesh", "Mesh", "mesh")
 
-    def execute(self, start, end, steps):
+    def draw(self, layout):
+        layout.prop(self, "lineMode", text = "")
+
+    def getExecutionFunctionName(self):
+        if self.lineMode == "STARTEND":
+            return "execute_StartEndLine"
+        elif self.lineMode == "POINTS":
+            return "execute_PointsLine"
+
+    def execute_StartEndLine(self, start, end, steps):
         steps = max(steps, 2)
-        return getLineMesh(start, end, steps)
\ No newline at end of file
+        return getLineMesh(start, end, steps)
+
+    def execute_PointsLine(self, points, cyclic):
+        if len(points) == 0: return Mesh()
+        return getLinesMesh(points, cyclic)
+        
```

Comment:
New line at end of file.

Reply:
Added.

Reply:
It seems you misunderstood me. Text files should always have a new line as the very last characters of the file. Editors typically do this automatically. You can setup VS Code to do that by adding the following to your config:
```
    "files.trimTrailingWhitespace": true,
    "files.insertFinalNewline": true,
    "files.trimFinalNewlines": true,
```

Also see https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file